spec/schema_tools/hash_spec.rb

```diff
@@ -217,6 +225,27 @@ class NestedObjectNoProperties
       hash['person']['first_name'].should == 'Pit'
     end
 
+    it 'has nested anyOf type object ' do
+      contact = Contact.new
+      contact.first_name = 'Pit'
+
+      i = AnyOfDefinition.new
+      i.person = contact
+
+      hash = SchemaTools::Hash.from_schema(i, exclude_root: true)
+      hash['person']['first_name'].should == 'Pit'
+    end
+
+    it 'has nested allOf type object ' do
+      contact = Contact.new
+      contact.first_name = 'Pit'
+
+      i = AllOfDefinition.new
+      i.person = contact
+
+      hash = SchemaTools::Hash.from_schema(i, exclude_root: true)
+      hash['person']['first_name'].should == 'Pit'
```

Comment:
Shouldn't we also check something specific to the `AllOfDefinition`? Like this key:

`"title":"test_allOf_object"`